packages/manager/modules/emailpro/src/service/messaging/emailpro.messaging.service.js

```diff
@@ -0,0 +1,169 @@
+import forEach from 'lodash/forEach';
+import isEmpty from 'lodash/isEmpty';
+import isString from 'lodash/isString';
+import uniqBy from 'lodash/uniqBy';
+
+export default class Messaging {
+  resetMessages() {
+    this.message = null;
+    this.messageDetails = null;
+  }
+
+  write(message, alertType, failure = null) {
+    this.message = message;
+    this.alertType = `alert ${alertType}`;
+    this.messageDetails = [];
+
+    if (failure != null) {
+      const value = [failure, failure.data];
+
+      forEach(value, (currentValue) => {
+        if (!isEmpty(currentValue)) {
+          if (isString(currentValue)) {
+            this.messageDetails.push({ id: null, message: currentValue });
+          } else if (isString(currentValue.message)) {
+            this.messageDetails.push({
+              id: currentValue.id,
+              message: currentValue.message,
+            });
+          } else if (currentValue.messages != null) {
+            this.messageDetails = currentValue.messages.map(
+              (currentMessage) => ({
+                id: currentMessage.code,
+                message: currentMessage.message,
+              }),
+            );
+            this.messageDetails = uniqBy(
+              this.messageDetails,
+              (currentMessage) =>
+                `${currentMessage.id}${currentMessage.message}`,
+            );
+          }
+        }
+      });
+    }
+  }
+
+  writeInfo(message, details = null) {
+    this.write(message, 'alert-info', details);
```

Comment:
css classnames are used more than once so you can create constants instead

Reply:
Do I need to refacto also this code from Exchange, to keep it homogeneous?